app/src/androidTest/java/tech/relaycorp/gateway/background/endpoint/EndpointPreRegistrationServiceTest.kt

```diff
@@ -7,50 +7,69 @@ import android.os.Looper
 import android.os.Message
 import android.os.Messenger
 import androidx.test.core.app.ApplicationProvider.getApplicationContext
+import androidx.test.platform.app.InstrumentationRegistry
 import androidx.test.rule.ServiceTestRule
+import kotlinx.coroutines.runBlocking
 import org.junit.Assert.assertEquals
 import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertTrue
 import org.junit.Rule
 import org.junit.Test
+import tech.relaycorp.gateway.data.disk.SensitiveStore
+import tech.relaycorp.gateway.domain.LocalConfig
+import tech.relaycorp.relaynet.messages.control.ClientRegistrationAuthorization
+import java.nio.charset.Charset
 
-class NonceRequestServiceTest {
+class EndpointPreRegistrationServiceTest {
 
     @get:Rule
     val serviceRule = ServiceTestRule()
 
     @Test
-    fun requestAndReceiveNonce() {
+    fun requestPreRegistration() = runBlocking {
         val serviceIntent = Intent(
             getApplicationContext<Context>(),
-            NonceRequestService::class.java
+            EndpointPreRegistrationService::class.java
         )
         val binder = serviceRule.bindService(serviceIntent)
 
         var resultMessage: Message? = null
 
         val messenger = Messenger(binder)
-        val requestMessage = Message.obtain(null, NonceRequestService.NONCE_REQUEST)
+        val requestMessage =
+            Message.obtain(null, EndpointPreRegistrationService.PREREGISTRATION_REQUEST)
         requestMessage.replyTo = Messenger(object : Handler(Looper.getMainLooper()) {
             override fun handleMessage(msg: Message) {
                 resultMessage = Message.obtain().also { it.copyFrom(msg) }
             }
         })
         messenger.send(requestMessage)
 
-        Thread.sleep(100)
+        Thread.sleep(500)
 
+        assertNotNull("We should have got a reply", resultMessage)
         assertEquals(
-            NonceRequestService.NONCE_RESULT,
-            resultMessage?.what
+            EndpointPreRegistrationService.REGISTRATION_AUTHORIZATION,
+            resultMessage!!.what
         )
-        assertNotNull(resultMessage?.obj)
+
+        // Check we got a valid CRA
+        assertTrue(resultMessage!!.obj is ByteArray)
+        val store = SensitiveStore(InstrumentationRegistry.getInstrumentation().targetContext)
+        val localConfig = LocalConfig(store)
```

Comment:
I'm bypassing dependency injection -- is there any way to just "ask" for a `localConfig` instance in this test?

Reply:
Makes sense, although it seems like Dagger wasn't fully integrated in the instrumentation tests, so I completed that and changed this code in b551c056722c36a8a5947b190230a940df3d8b23. It'd be good to get your feedback there too.